core/modules/file/file.module

```diff
@@ -2795,20 +2795,21 @@ function file_get_type($file) {
  *   TRUE if the operation may be performed, FALSE otherwise.
  */
 function file_access($op, $file = NULL, $account = NULL) {
+  // Filter out unsupported operations.
   if (!in_array($op, array('create', 'view', 'download', 'update', 'delete'), TRUE)) {
     return FALSE;
   }
 
-  // If neither a file object nor a file type are given, we make sure that the
-  // account has access regardless of bundle.
-  if (!$file) {
-    $bypass_access = user_access('bypass file access', $account);
-
+  // If no $file is given or it's only the file type.
+  if (!$file || is_string($file)) {
+    // Filter out unsupported file types.
+    if (is_string($file) && !in_array($file, file_permissions_get_configured_types())) {
+      return FALSE;
+    }
     if ($op == 'create') {
```

Comment:
This leaves out ops other than 'create' when $file  is a string (file type) or null.  Don't we want the ability to check also in those cases? (for example checking if a user has permission to download any file of type, or of a specific type?)

Reply:
Further, at the moment it's not possible to check if a user has the global permission 'view files' if not file or file type is provided. This is a bit counterintuitive. The permission 'view files' is global, so one would assume you would not need to provide a file object or file type...

Reply:
There's no other way to do this.

For other checks it's impossible to do anything useful without an actual instance. Think of the checks re "own", or private files. This logic isn't really different from previous behavior.

Reply:
We can do a little more here, but still not everything would be possible without an instance.